### PR TITLE
Digital view of granted licence

### PR DESCRIPTION
**Pull request by joefitter**
* Updated protocols to include table
* added granted only sections
* added styles for step index interface

### CODE REVIEW

File: client/components/review.js [hunk 1]
```diff
@@ -28,13 +29,13 @@ class Review extends React.Component {
       value = options.find(option => !isUndefined(option.value) ? option.value === value : option === value)
     }
 
-    if (value && this.props.type === 'duration') {
+    if (this.props.type === 'duration') {
       return (
         <dl className="inline">
           <dt>Years</dt>
-          <dd>{value.years || 0}</dd>
+          <dd>{(value || {}).years || 5}</dd>
```
Comment (wheelsandcogs) [inline client/components/review.js]: Does this default to 5 years if it's not set?

Comment (joefitter) [reply client/components/review.js]: there's some code in resolver which defaults it to 5 years

File: client/pages/section.js [hunk 2]
```diff
@@ -81,7 +70,12 @@ class Section extends React.Component {
               data = { [data]: value };
             }
             const conditions = getConditions({ ...this.props.project, ...data }, CONDITIONS.project);
-            return this.props.update({ ...data, conditions });
+
+            this.props.update(data);
+
+            if (!this.props.isLegacy) {
+              this.props.update({ conditions })
```
Comment (wheelsandcogs) [inline client/pages/section.js]: ;

File: client/pages/sections/granted/project-summary.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    licenceNumber,
+    address,
+    licenceHolder: {
+      firstName,
+      lastName
+    }
+  },
+  fields
+}) => {
+  const ra = (values.conditions || []).find(c => /retrospective-assessment/.test(c.key));
```
Comment (wheelsandcogs) [inline client/pages/sections/granted/project-summary.js]: Saving those precious chars? 😄

File: client/pages/sections/granted/project-summary.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      <div className="granted-section">
+        <Review
+          {...fields.find(f => f.name === 'duration')}
+          value={values.duration}
+          noComments
+        />
+      </div>
+      <div className="granted-section">
+        <h3>Date granted</h3>
+        <p>{format(project.issueDate, 'DD MMMM YYYY')}</p>
```
Comment (wheelsandcogs) [inline client/pages/sections/granted/project-summary.js]: Might be worth moving these date format strings into their own file somewhere like we do in pages (or even better, somewhere shared across all projects like `asl-constants`)

File: client/pages/section.js (hunk 2 above)
Comment (joefitter) [reply client/pages/section.js]: do you remember that time when I added semicolons to lint, and then ran `lint --fix`, and then removed it, and then was going to add it back? Yeah that. I wont do it here, but I'll submit a separate PR to sweep all of them

File: client/pages/sections/granted/project-summary.js (hunk 4 above)
Comment (joefitter) [reply client/pages/sections/granted/project-summary.js]: This project doesn't use our artifactory npm registry so has no access to asl-constants else I would have just used that. I do agree these magic strings should be defined somewhere better though - please hold.

File: client/pages/sections/granted/project-summary.js (hunk 3 above)
Comment (joefitter) [reply client/pages/sections/granted/project-summary.js]: always x

File: client/pages/sections/protocols/protocol-sections.js [hunk 5]
```diff
@@ -43,29 +43,67 @@ class ProtocolSections extends PureComponent {
...
     const severityField = sections.details.fields.find(field => field.name === 'severity');
-    const severityOption = severityField.options && severityField.options.find(option => option.value === values.severity);
+    const severityOption = (severityField.options.find(option => option.value === values.severity) || {}).label;
 
     const numberOfNewComments = Object.values(newComments)
       .reduce((total, comments) => total + (comments || []).length, 0);
 
+    const speciesDetails = values.speciesDetails && values.speciesDetails.filter(s => s.name);
+
+    const na = <em>No answer provided</em>;
```
Comment (wheelsandcogs) [inline client/pages/sections/protocols/protocol-sections.js]: Come on, lets have `noAnswer` please :)

Comment (joefitter) [reply client/pages/sections/protocols/protocol-sections.js]: sorry, I was taking a leaf out of @lvnhmd 's book :trollface:

I'm actually shocked I wrote that.

sorrybox.gif

**Pull request merged**